qiskit/assembler/assemble_schedules.py

```diff
@@ -53,21 +55,26 @@ def assemble_schedules(schedules, qobj_id, qobj_header, run_config):
 
     # Pack everything into the Qobj
     qobj_schedules = []
-    user_pulselib = set()
+    user_pulselib = {}
     for idx, schedule in enumerate(schedules):
         # instructions
         qobj_instructions = []
         # Instructions are returned as tuple of shifted time and instruction
         for shift, instruction in schedule.instructions:
             # TODO: support conditional gate
-            qobj_instructions.append(instruction_converter(shift, instruction))
             if isinstance(instruction, PulseInstruction):
+                name = instruction.command._name
+                if name in user_pulselib and not np.allclose(instruction.command.samples,
+                                                             user_pulselib[name]):
+                    name = "{0}-{1:x}".format(name, hash(instruction.command.samples.tostring()))
                 # add samples to pulse library
-                user_pulselib.add(instruction.command)
+                user_pulselib[name] = instruction.command.samples
```

Comment:
If you take the approach above, I guess you would need to first create a user_pulselib of `name: SamplePulse` pairs, and then unwrap this later to a new pulse library of `name: array` pairs. I think you should do what you feel is best.